modules/saml/www/sp/saml2-logout.php

```diff
@@ -12,11 +12,11 @@
 
 $sourceId = substr($_SERVER['PATH_INFO'], 1);
 
+/** @var \SimpleSAML\Module\saml\Auth\Source\SP $source */
 $source = \SimpleSAML\Auth\Source::getById($sourceId);
 if ($source === null) {
     throw new \Exception('Could not find authentication source with id '.$sourceId);
-}
-if (!($source instanceof \SimpleSAML\Module\saml\Auth\Source\SP)) {
+} elseif (!($source instanceof \SimpleSAML\Module\saml\Auth\Source\SP)) {
```

Comment:
The same as with state arrays happen here. If it is actually possible that the auth source is `null`, then we should check for that first, then assert that it's an object of that class in the phpdoc.

Reply:
Unfortunately annotations only work on assignments.. We would have to use assert() to go with your approach

Reply:
Uhm, I was pretty sure I tried using annotations at random places in the code and that worked fine, but I might be wrong... 🤔

Reply:
He're proof: https://psalm.dev/r/262a290058

Reply:
Interesting! Also for the interactive tool that I didn't know!

Reply:
Yeah, it's great! I think it's fairly new..
They will ask you to use it if you report a bug so that they can rule out that the bug still exists in the latest version and so they have means of reproducing a bug... Really nice!